src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -2138,6 +2138,23 @@ public void GetAsync_ServerNeedsAuthAndNoCredential_StatusCodeUnauthorized()
             });
         }
 
+        [OuterLoop]
+        [Fact]
+        public async Task GetAsync_IdnHostName_SuccessStatusCodeInResponse()
```

Comment:
I would change this name to:  "GetAsync_UnicodeHostName_SuccessStatusCodeInResponse"

Reply:
Also, did you confirm that this test will fail without the fix in WinHttpHandler (to use IdnHost instead of Host)?

Reply:
actually no, i did only IdnHost square brackets tests. I'll do test.

Reply:
>  Note that WinHttp does not accept international host names without converting them first to Punycode

@davidsh...weird...seems works, i tried also others url to test different characters...

Reply:
>@davidsh...weird...seems to work, i tried also others url to test different characters...

Perhaps WinHTTP has changed behavior on recent Windows OS versions to allow Unicode characters to be used in the API calls.  However, if you look at WireShark traces, you'll probably see IDN (punnycode) encoding actually used underneath.  Did you observe that?

It's possible that the documentation is out of date and needs to be corrected.  This needs to be researched.  @caesar1995 can you start an offline email with WinHTTP team and ask them about this behavior?

For now, it would be useful to update your PR to remove the product code change in WinHttpHandler but keep the new test you added.  Then run all the tests against this PR and see what passes/fails.  Perhaps older Windows OS versions like Windows 7 might fail.  We need to understand this more.  It's possible that we don't need a product code change at all but the new test is useful.

Reply:
i did a test with fiddler and i see punycode going out.